### PR TITLE
 Add api endpoints list of indicators, reports, layers, datasets and fid_field

### LINKED ISSUE #106: Add API endpoints to retrive a list of indicators, reports, layers, datasets and fid_field
The API should have the same functionality as the CLI. See cli.py.

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description
Add api-endpoints list of indicators, reports, layers, datasets and fid_field and tests for them.

### Corresponding issue
Closes #106 

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

Please check all finished tasks. If some tasks do not apply to your PR, please cross their text out (by using `~...~`) and remove their checkboxes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,13 +26,15 @@
 - Minor improvements to the metadata and docstrings of indicators ([#110])
 - Change return type from bool to None for indicator for functions: preprocess, calculate, create_figure ([#96])
 - Implement mapping of custom feature id to default (unique) feature id ([#83])
+- Add API-endpoints for list of indicators, reports, layers, datasets and fid_field ([#106])
```
Comment (joker234) [inline CHANGELOG.md:29]: isn't this a new feature?

File: workers/ohsome_quality_analyst/api.py [hunk 2]
```diff
@@ -149,4 +154,53 @@ async def _fetch_report(
...
+    response = empty_api_response()
+    response["result"] = list(load_metadata("indicators").keys())
+    return response
+
+
+@app.get("/datasetNames")
+async def list_datasets():
+    """List names of available datasets."""
+    response = empty_api_response()
+    response["result"] = list(list(DATASETS.keys()))
```
Comment (matthiasschaub) [inline workers/ohsome_quality_analyst/api.py:176]: Is it intentional that `list()` is called two times?

```suggestion
    response["result"] = list(DATASETS.keys())
```

File: workers/tests/integrationtests/test_api_regions.py [hunk 3]
```diff
@@ -21,6 +21,41 @@ def test_get_available_regions(self):
...
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(geojson.loads(response.content), dict)
+        self.assertIsNotNone(geojson.loads(response.content))
+
+    def test_get_list_reports(self):
+        url = "/reportNames"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(geojson.loads(response.content), dict)
+        self.assertIsNotNone(geojson.loads(response.content))
```
Comment (matthiasschaub) [inline workers/tests/integrationtests/test_api_regions.py:57]: Testing for None is not necessary anymore after testing if the object is of instance dict:

```suggestion
    def test_get_list_indicators(self):
        url = "/indicatorNames"
        response = self.client.get(url)
        self.assertEqual(response.status_code, 200)
        self.assertIsInstance(geojson.loads(response.content), dict)

    def test_get_list_layers(self):
        url = "/layerNames"
        response = self.client.get(url)
        self.assertEqual(response.status_code, 200)
        self.assertIsInstance(geojson.loads(response.content), dict)

    def test_get_list_datasets(self):
        url = "/datasetNames"
        response = self.client.get(url)
        self.assertEqual(response.status_code, 200)
        self.assertIsInstance(geojson.loads(response.content), dict)

    def test_get_list_fid_field(self):
        url = "/FidFields"
        response = self.client.get(url)
        self.assertEqual(response.status_code, 200)
        self.assertIsInstance(geojson.loads(response.content), dict)

    def test_get_list_reports(self):
        url = "/reportNames"
        response = self.client.get(url)
        self.assertEqual(response.status_code, 200)
        self.assertIsInstance(geojson.loads(response.content), dict)
```

**Review by matthiasschaub — COMMENTED** [review]
I left some minor comments and change suggestions.
Also I would rename the file `workers/tests/integrationtests/test_api_regions.py` to  workers/tests/integrationtests/test_api.py` since it not only tests `list_regions` anymore.

File: workers/ohsome_quality_analyst/api.py [hunk 4]
```diff
@@ -148,5 +153,67 @@ async def _fetch_report(
 
 
 @app.get("/regions")
-async def get_available_regions():
-    return await db_client.get_available_regions()
+async def get_available_regions(
+    request: Request,
+):
```
Comment (matthiasschaub) [inline workers/ohsome_quality_analyst/api.py:158]:
```suggestion
async def get_available_regions(request: Request):
```

**Review by matthiasschaub — COMMENTED** [review]
@Gigaszi  Great work, thanks. This PR looks good to me. Fell free to go ahead and merge. I suggest using the `squash and merge` option. First though you need to approve the pull request since I am not able to (I opened the PR).

**Review by Gigaszi — APPROVED** [review]

**Pull request merged**